eldritch/test_encounters.py

```diff
@@ -1566,5 +1567,527 @@ def testIsle7Fail(self):
+      self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 1)
+    self.assertEqual(self.char.place.name, "Asylum")
+    self.assertEqual(self.char.clues, 1)
+
+  def testGraveyard2Fail(self):
+    self.state.event_stack.append(encounters.Graveyard2(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=4)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Rivertown")
+
+  def testGraveyard3Pass(self):
+    self.char.fight_will_slider = 2
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.clues, 1)
+    self.assertEqual(len(self.char.possessions), 1)
+    self.assertEqual(self.char.possessions[0].name, "Holy Water")
+
+  def testGraveyard3FailOne(self):
+    self.char.fight_will_slider = 2
+    self.char.stamina = 4
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=1)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 3)
+
+  def testGraveyard3FailTwo(self):
+    self.char.fight_will_slider = 2
+    self.char.stamina = 4
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=2)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 2)
+
+  def testGraveyard3FailThree(self):
+    self.char.fight_will_slider = 2
+    self.char.stamina = 4
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=3)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 1)
+
+  def testGraveyard3FailFour(self):
+    self.char.fight_will_slider = 2
+    self.char.stamina = 5
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=4)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 1)
+
+  def testGraveyard3FailKO(self):
+    self.char.fight_will_slider = 2
+    self.char.stamina = 4
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=4)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Hospital")
+    self.assertEqual(self.char.stamina, 1)
+
+#  def testGraveyard4NoTrophies(self):
```

Comment:
These seem less critical to have a todo for when the event itself is entirely unimplemented.